tools/c7n_azure/tests/test_cost_management_export.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+from collections import namedtuple
+
+from azure_common import BaseTest, arm_template, cassette_name
+from mock import patch
+
+
+class CostManagementExportTest(BaseTest):
+
+    class MockExecutionHistory:
+        def __init__(self, submitted_time_list):
+            self.value = []
+            MockExecutionItem = namedtuple('MockExecutionItem', ['submitted_time', 'serialize'])
+            for t in submitted_time_list:
+                self.value.append(MockExecutionItem(submitted_time=t, serialize=lambda b: ''))
+
+    def test_key_vault_storage_schema_validate(self):
+        p = self.load_policy({
+            'name': 'cost-management-export',
+            'resource': 'azure.cost-management-export',
+            'filters': [
+                {'type': 'last-execution',
+                 'age': 30}
+            ],
+            'actions': [
+                {'type': 'execute'}
+            ]
+        }, validate=True)
+        self.assertTrue(p)
+
+    @arm_template('cost-management-export.json')
+    @cassette_name('common')
+    def test_cost_management_export(self):
+        p = self._get_policy()
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+    # If we use 0 for 'age', we should get back same list of exports
+    # 1. If it was executed right now, it still <= 0 days ago
+    # 2. Exports with no executions are always included in the filter
+    # This test is primarily to catch SDK changes regressions
+    @arm_template('cost-management-export.json')
+    @cassette_name('last-execution')
+    def test_cost_management_export_last_execution(self):
+        p = self._get_policy(filters=[{'type': 'last-execution', 'age': 0}])
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+    # There is no guarantee we have or don't have some real execution, so we will simulate possible
+    # scenarios using patch
+    @patch('azure.mgmt.costmanagement.operations.ExportsOperations.get_execution_history',
+           return_value=MockExecutionHistory([datetime.datetime.now()]))
+    @arm_template('cost-management-export.json')
+    @cassette_name('common')
+    def test_cost_management_export_last_execution_mock(self, _1):

```

Comment:
(minor) The naming of the tests here are a bit confusing. It's unclear what the outcome is suppose to be, because one talks about the result (no_results) and another one is talking about the input (no_executions).